ec2/vpc/vpc.tf

```diff
@@ -69,12 +75,21 @@ resource "aws_route" "public" {
   destination_cidr_block = "0.0.0.0/0"
 }
 
+resource "aws_route" "ip6-public" {
+  depends_on                  = ["aws_internet_gateway.public", "aws_route_table.public"]
+  route_table_id              = "${aws_route_table.public.id}"
+  gateway_id                  = "${aws_internet_gateway.public.id}"
+  destination_ipv6_cidr_block = "::/0"
+}
+
 resource "aws_subnet" "public" {
-  count                   = "${local.az_count}"
-  vpc_id                  = "${aws_vpc.main.id}"
-  cidr_block              = "${cidrsubnet(var.cidr_block, local.az_count + local.private_count, count.index)}"
-  availability_zone       = "${element(data.aws_availability_zones.main.names, count.index)}"
-  map_public_ip_on_launch = "true"
+  count                           = "${local.az_count}"
+  vpc_id                          = "${aws_vpc.main.id}"
+  cidr_block                      = "${cidrsubnet(var.cidr_block, local.az_count + local.private_count, count.index)}"
+  availability_zone               = "${element(data.aws_availability_zones.main.names, count.index)}"
+  map_public_ip_on_launch         = "true"
+  ipv6_cidr_block                 = "${cidrsubnet(aws_vpc.main.ipv6_cidr_block, 8, count.index)}"
```

Comment:
This will give every subnet 1/8th of the IPV6 CIDR block? Is it possible to use `local.az_count + local.private_count` as done for IPV4?

Reply:
Same for the private subnets.

Reply:
I did origianlly try it that way but got error messages saying invalid subnet size.  My understanding is that  Amazon's implementation of IPv6 mandates that the vpc gets a /54 and the subnets must be a /64  so 8 is the only valid entry. I think this gives you get 2<sup>8</sup> subnets
Here is a good post on what newbits and netnum for cidrsubnet is about http://blog.itsjustcode.net/blog/2017/11/18/terraform-cidrsubnet-deconstructed/

Reply:
Doesn't 2^8 seem very low? I mean I can definitely see us go above 256 subnets. Not that we can do anything about it if AWS doesn't allow other ranges but still.

Reply:
I see now that this is per VPC, don't think we will have more then 256 subnets per VPC :)

Reply:
Hehe, think the limit is 200 subnets per vpc also ;)

Reply:
@colincoleman - Thanks for the link! It appears I have not understood that function at all 😆 This obviously needs fixing for IPV4 also, should we just fix it in this PR?

Reply:
>My understanding is that Amazon's implementation of IPv6 mandates that the vpc gets a /54 and the subnets must be a /64 so 8 is the only valid entry. I think this gives you get 28 subnets.

Wouldn't a `::/64` subnet give you 256 IPV6's in the subnet?

Documentation for the requirement (FYI): https://www.terraform.io/docs/providers/aws/r/subnet.html
> (Optional) The IPv6 network range for the subnet, in CIDR notation. The subnet size must use a /64 prefix length.

Reply:
Oh, I really don't understand IPV6 - from an online calculator it seems like `::1/64` is `18446744073709551616` IPs. We sure that is enough?

Reply:
Ok, if `2^64/2^54 = 1024` we can have 1024 subnets?

Reply:
Sorry typo above - /56 is the VPC

Reply:
ISPs are handing out /64 addresses to residential customers  and bigger blocks to business...

Reply:
That is one _very_ smart home...

Reply:
Should we do a separate PR for IPV4 or do you think we can figure out a better solution and include it in this one?

Reply:
I'll take a look at it now and include it here - that was less to merge when we move to telia-oss